expo-location crashes with Unable to resume activity

### Summary

https://forums.expo.dev/t/unable-to-resume-activity/63114

I've decided to create this here, hoping for some kind of answer, since the expo forums are a no-go, most of the time.

The bug is surely with the expo-location module, and not with the app, since there are multiple users having the same issue, and the crash occurs thousands of times on our customer's devices, as we have about ~50K devices with the app installed, so it affects quite a lot of people.

I've tried describing it into the forum thread, so I'm not sure if I should paste it here as well, but as a summary, this seems to happen when you leave the app in the background during the night, and open it in the morning, since it mostly occurs during the morning, as another developer mentioned on the forum.

```
Caused by java.lang.NullPointerException
       at java.util.Objects.requireNonNull(Objects.java:220)
       at expo.modules.location.LocationModule.handleForegroundLocationPermissions(LocationModule.java:839)
       at expo.modules.location.LocationModule.lambda$requestForegroundPermissionsAsync$2$LocationModule(LocationModule.java:195)
       at expo.modules.location.-$$Lambda$LocationModule$D--6Q7JrlKjA_yBo0ZZZH8hwBGk.onResult(:4)
       at org.unimodules.adapters.react.permissions.PermissionsService$createListenerWithPendingPermissionsRequest$1.onRequestPermissionsResult(PermissionsService.kt:255)
       at com.facebook.react.ReactActivityDelegate$2.invoke(ReactActivityDelegate.java:171)
       at com.facebook.react.ReactActivityDelegate.onResume(ReactActivityDelegate.java:102)
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android, iOS

### SDK Version (managed workflow only)

44 and 45

### Environment

This is applicable to all devices .. not environment dependent

### Reproducible demo

It's hard to reproduce, but the stack traces from all the users mentioning the issue I guess should be enough.

@byCedric I don't seem to have it installed anywhere in the project.

```
edi@MBP byspark % npm list react-native-unimodules
BySpark@ /Users/edi/Desktop/BySpark
└── (empty)
```

Here's one of my crash logs, as the one I pasted was from the last guy who replied to my thread on the forum, as I assumed it's the same.

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4638)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4670)
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:52)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2253)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:245)
  at android.app.ActivityThread.main (ActivityThread.java:8024)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:631)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:978)
Caused by: java.lang.reflect.InvocationTargetException: 
  at java.lang.reflect.Method.invoke (Native Method)
  at expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod (ReactActivityDelegateWrapper.kt:166)
  at expo.modules.ReactActivityDelegateWrapper.onResume (ReactActivityDelegateWrapper.kt:82)
  at com.facebook.react.ReactActivity.onResume (ReactActivity.java:58)
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1571)
  at android.app.Activity.performResume (Activity.java:8141)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4628)
Caused by: java.lang.NullPointerException: 
  at java.util.Objects.requireNonNull (Objects.java:220)
  at expo.modules.location.LocationModule.handleForegroundLocationPermissions (LocationModule.java:839)
  at expo.modules.location.LocationModule.lambda$requestForegroundPermissionsAsync$2$LocationModule (LocationModule.java:195)
  at expo.modules.location.-$$Lambda$LocationModule$BTuImArYyLs9EqpIXxluCGJIymU.onResult (Unknown Source:4)
  at expo.modules.adapters.react.permissions.PermissionsService$createListenerWithPendingPermissionsRequest$1.onRequestPermissionsResult (PermissionsService.kt:271)
  at com.facebook.react.ReactActivityDelegate$2.invoke (ReactActivityDelegate.java:178)
  at com.facebook.react.ReactActivityDelegate.onResume (ReactActivityDelegate.java:103)
```

--- a/packages/expo-location/CHANGELOG.md
+++ b/packages/expo-location/CHANGELOG.md
@@ -8,6 +8,7 @@
 
 ### 🐛 Bug fixes
 
+- Removed strict null checks for expo location and avoid crash on android. ([#20792](https://github.com/expo/expo/pull/20792) by [@jayshah123](https://github.com/jayshah123) and [@forki](https://github.com/forki))
 - Export types with type-only annotation to fix build when using `isolatedModules` flag. ([#20239](https://github.com/expo/expo/pull/20239) by [@zakharchenkoAndrii](https://github.com/zakharchenkoAndrii))
 
 ### 💡 Others

--- a/packages/expo-location/android/src/main/java/expo/modules/location/LocationModule.java
+++ b/packages/expo-location/android/src/main/java/expo/modules/location/LocationModule.java
@@ -841,8 +841,15 @@ public class LocationModule extends ExportedModule implements LifecycleEventList
   private Bundle handleForegroundLocationPermissions(Map<String, PermissionsResponse> result) {
     PermissionsResponse accessFineLocation = result.get(Manifest.permission.ACCESS_FINE_LOCATION);
     PermissionsResponse accessCoarseLocation = result.get(Manifest.permission.ACCESS_COARSE_LOCATION);
-    Objects.requireNonNull(accessFineLocation);
-    Objects.requireNonNull(accessCoarseLocation);
+    /**
+     * Missing permissions from OS callback should be considered as denied permissions
+     */
+    if(accessFineLocation == null) {
+      accessFineLocation = new PermissionsResponse(PermissionsStatus.DENIED, true);
+    }
+    if(accessCoarseLocation == null) {
+      accessCoarseLocation = new PermissionsResponse(PermissionsStatus.DENIED, true);
+    }
 
     PermissionsStatus status = PermissionsStatus.UNDETERMINED;
     String accuracy = "none";

